pkg/controller/cluster/cluster_reconcile.go

```
@@ -93,10 +107,27 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 			return reconcile.Result{}, err
 		}
 
-		err = client.SetToken(path.Join(instance.Spec.TenantRef.Name, instance.Name, "steward"), token, reqLogger)
+		err = vault.SetToken(secretPath, token)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
+
+		vault.SetDeletePath(secretPath)
+
+		deleted, err := helpers.HandleDeletion(instance, vault, finalizerName, r.client)
```

This error is never handled, if `deleted` is `true`